File: K/2154-keep-multiplying-found-values-by-two/main.go
package main

import "fmt"

func findFinalValue(nums []int, original int) int {
	m := map[int]int{}
	for _, v := range nums {
		m[v] = 1
	}

	o := original
	for {
		_, ok := m[o]
		if ok {
			o *= 2
		} else {
			return o
		}
	}
}

func main() {
	fmt.Println()
}

File: K/2154-keep-multiplying-found-values-by-two/main.rs
use std::collections::HashSet;

impl Solution {
  pub fn find_final_value(nums: Vec<i32>, original: i32) -> i32 {
    let mut m: HashSet<i32> = HashSet::from_iter(nums.iter().cloned());
    let mut o = original;
    while m.contains(&o) {
      o *= 2;
    }
    o
  }
}
